src/textual/message_pump.py

```diff
@@ -578,7 +578,16 @@ def _get_dispatch_methods(
             # Try decorated handlers first
             decorated_handlers = cls.__dict__.get("_decorated_handlers")
             if decorated_handlers is not None:
-                handlers = decorated_handlers.get(type(message), [])
+                handlers: list[tuple[Callable, dict[str, tuple[SelectorSet, ...]]]] = []
+                add_handlers = handlers.extend
+                # Allow for firing of handlers bound to a message higher up
+                # the inheritance tree.
+                for check_message in message.__class__.__mro__:
+                    # If we've hit something that isn't derived from Message
+                    # we can give up.
+                    if not issubclass(check_message, Message):
```

Comment:
This may fail in the case of multiple inheritance or mixins. I would hope that nobody uses those in the context of messages, but maybe we should continue rather than break.

Reply:
Fair point, I'll add some testing for this first. The intent here was to not go further up the MRO than `Message`, to save looping more than necessary (although of course, at the moment, that's only really going to save one more trip round).

Reply:
Test added and the "get out as soon as possible" code changed to "wander up the MRO and check anything that looks like a `Message`): 02b862c28c57fff6186a35a1ed611cd1d0f90860